### PR TITLE
Add installation instructions

### PR DESCRIPTION
**Pull request by goderbauer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +15,96 @@ Not all features of the API are implemented in the plugin yet. If something is m
...
+
+You can find the most recent version of the plugin on [pub](https://pub.dartlang.org/packages/firebase_messaging).
+
+### Android Integration
+
+To integrate your plugin into the Android part of your app, follow these steps:
+
+1. Using the [Firebase Console](https://console.firebase.google.com/) add an Android app to your project: Follow the assistant, download the generated `google-services.json` file and place it inside `android/app`. Next, modify the `android/build.gradle` file and the `android/app/build.gradle` file to add the Google services plugin as described by the Firebase assistant.
+
+1. Include the following `intent-filter` within the `<activity>` tag of your `android/app/src/main/AndroidManifest.xml`:
```
Comment (collinjackson) [inline README.md]: We decided this is optional, right? Maybe mention that?

File: README.md [hunk 2]
```diff
@@ -14,7 +15,96 @@ Not all features of the API are implemented in the plugin yet. If something is m
...
+   ```
+1. Override the `onNewIntent` method in your `android/app/src/main/java/.../MainActivity.java` to forward the call to the plugin:
+   ```java
+   import android.content.Intent;
+   
+   // ...
+   
+   @Override
+   protected void onNewIntent(Intent intent) {
+     pluginRegistry.firebase_messaging.onNewIntent(intent);
```
Comment (collinjackson) [inline README.md]: My dream is that someday FlutterActivity will forward this to the plugin automatically, but it's fine for now.

File: README.md [hunk 3]
```diff
@@ -14,7 +15,96 @@ Not all features of the API are implemented in the plugin yet. If something is m
...
+
+```objective-c
+- (BOOL)application:(UIApplication *)application
+  // ...
+  [plugins.firebase_messaging didFinishLaunchingWithOptions:launchOptions];
+  // ...
+}
+
+- (void)applicationDidEnterBackground:(UIApplication *)application {
+  [plugins.firebase_messaging applicationDidEnterBackground];
```
Comment (collinjackson) [inline README.md]: And same with this one (flutter/flutter#9682), hopefully someday we won't have to do this step, but it's fine for now.

File: README.md (hunk 2 above)
Comment (goderbauer) [inline README.md]: +1 to that.

File: README.md (hunk 3 above)
Comment (goderbauer) [inline README.md]: +1 to that as well. Can't wait for it.

File: README.md (hunk 1 above)
Comment (goderbauer) [inline README.md]: Clarified that it's optional.

**Pull request merged**